devnet/cross-client/docker-compose.yml

```diff
@@ -0,0 +1,49 @@
+services: 
+  alice:
+    platform: linux/amd64
+    build:
+      context: ./../..
+      dockerfile: devnet/alice.Dockerfile
+      args:
+        DD_API_KEY: ${DD_API_KEY}
+    ports:
+      - 7001
+      - 8545
+      - 8546
+      - 8540
+      - 9876
+      - 6060
+
+  bob:
+    platform: linux/amd64
+    build:
+      context: ./../..
+      dockerfile: devnet/bob.Dockerfile
+      args:
+        key: bob
+        DD_API_KEY: ${DD_API_KEY}
+    ports:
+      - 7001
+      - 8545
+      - 8546
+      - 8540
+      - 9876
+      - 6060
```

Comment:
If you do want to publish ports on the host, then change them to a mapping. This will just port map them to random available ports on the host, I don't really see the use here.

For example:

```yml
version: '3'

services:
    alpine:
        image: alpine
        init: true
        command: [ "sleep", "100" ]
        ports:
            - 8000
```

```sh
docker-compose up -d
docker ps | grep "8000"
```

Will show

```
5118bd0982d9   alpine                         "sleep 100"              2 seconds ago   Up 2 seconds          0.0.0.0:49154->8000/tcp, :::49154->8000/tcp     workspace-alpine-1
```